packages/inventory-compliance/src/SystemRulesTable/SystemRulesTable.js

```diff
@@ -259,7 +259,8 @@ class SystemRulesTable extends React.Component {
 
         let selectedRuleIds;
         if (this.selectAllCheckbox(key)) {
-            selectedRuleIds = this.getRules().map((rule) => rule.refId);
+            let ruleRefIds = this.getRules().map((rule) => rule.refId);
+            selectedRuleIds = selected ? [ ...new Set([ ...selectedRefIds, ...ruleRefIds ]) ] : selectedRefIds.filter((refId) => !ruleRefIds.includes(refId));
```

Comment:
Can you please format the long line? :slightly_smiling_face:

Reply:
On System Details page rules one you select all the deselect all does nothing (no set change, checkbox stays at checked).

Reply:
I can try to add the system details page in this PR as well.